visionparams:
    ball_fcnn:
        model_path: '/models/fcnn03'
        debug: true  # only active when true and vision/debug is true, too!
        threshold: .5  # minimal value for a candidate to be considered
        expand_stepsize: 4
        pointcloud_stepsize: 10
        shuffle_candidate_list: true  # shuffles the list of possible candidate points
        min_ball_diameter: 5
        max_ball_diameter: 200

    field_color_detector:
        path: '/config/color_spaces/Iran_Feld_nah.yaml'
        use_dummy_green: false
        dummy_lower_values: [0, 0, 180]
        dummy_upper_values: [0, 0, 180]

    white_color_detector:
        lower_values: [0, 0, 180]
        upper_values: [255, 140, 255]

    horizon_finder:
        vertical_steps: 30
        horizontal_steps: 30
        precision_pix: 5
        min_precision_pix: 3

    vision:
        debug: true
        ball_candidate_horizon_y_offset: 0
        ball_candidate_rating_threshold: 0.5

    line_detector:
        horizon_offset: 50
        linepoints_range: 2000
        blur_kernel_size: 9

    ROS:
        # Subscriber
        img_msg_topic: 'image_raw'
        img_queue_size: 1
        # Publisher
        ball_msg_topic: 'ball_in_image'
        line_msg_topic: 'line_in_image'

    ball_finder:
        classify_threshold: 0.5
        scale_factor: 1.1
        min_neighbors: 1
        min_size: 10

    classifier:
       model_path: '/models/classifier_01'

    cascade_classifier:
       path: '/classifier/cascadeNew.xml'
